src/array/cuda/array_index_select.cu

```diff
@@ -63,18 +67,30 @@ template NDArray IndexSelect<kDLGPU, double, int64_t>(NDArray, IdArray);
 template <DLDeviceType XPU, typename DType>
 DType IndexSelect(NDArray array, int64_t index) {
   auto device = runtime::DeviceAPI::Get(array->ctx);
+#ifdef USE_FP16
+  // The initialization constructor for __half is apparently a device-
+  // only function in some setups, but the current function isn't run
+  // on the device.
```

Comment:
I am not clear which 'current' function we are talking here.

Reply:
"the current function" here refers to the function containing this comment, the function that is currently being run (on the host) as this comment is passed.  If I referred to it by name instead, i.e. `IndexSelect`, it might sound like the comment is referring to the other function named `IndexSelect`, above.  Maybe it would be clear if I included both, though. 🤔  I'll try something when I add the "TODO"s.

Reply:
I updated the comment.  Hopefully it's a bit clearer now.  Thanks!